tests/lib/nested.sh

```diff
@@ -268,99 +268,176 @@ cleanup_nested_env(){
     rm -rf "$WORK_DIR"
 }
 
-create_nested_core_vm(){
-    mkdir -p "$WORK_DIR/image"
-    if [ ! -f "$WORK_DIR/image/ubuntu-core.img" ]; then
-        local UBUNTU_IMAGE
-        UBUNTU_IMAGE=/snap/bin/ubuntu-image
-
-        # create ubuntu-core image
-        local EXTRA_FUNDAMENTAL=""
-        local EXTRA_SNAPS=""
-        if [ -d "${PWD}/extra-snaps" ]; then
-            while IFS= read -r mysnap; do
-                EXTRA_SNAPS="$EXTRA_SNAPS --snap $mysnap"
-            done <   <(find extra-snaps -name '*.snap')
-        fi
+get_image_path(){
+    echo "$WORK_DIR/image"
+}
+
+get_classic_image_name(){
+    get_image_name classic
+}
+
+get_image_name(){
+    local TYPE="$1"
+    local SOURCE="${CORE_CHANNEL}"
+    local VERSION="16"
+
+    if is_core_20_nested_system; then
+        VERSION="20"
+    elif is_core_18_nested_system; then
+        VERSION="18"
+    fi
+
+    if [ "$BUILD_SNAPD_FROM_CURRENT" = "true" ]; then
+        SOURCE="custom"
+    fi
+    if [ "$(get_extra_snaps | wc -l)" != "0" ]; then
+        SOURCE="custom"
+    fi
+    echo "ubuntu-${TYPE}-${VERSION}-${SOURCE}.img"
+}
+
+prepare_image_dir(){
+    mkdir -p "$(get_image_path)"
+}
+
+get_extra_snaps_path(){
+    echo "${PWD}/extra-snaps"
+}
 
-        local NESTED_MODEL=""
-        case "$SPREAD_SYSTEM" in
+get_extra_snaps(){
+    local EXTRA_SNAPS=""
+    local EXTRA_SNAPS_PATH
+    EXTRA_SNAPS_PATH="$(get_extra_snaps_path)"
+
+    if [ -d "$EXTRA_SNAPS_PATH" ]; then
+        while IFS= read -r mysnap; do
+            echo "$mysnap"
+        done <   <(find "$EXTRA_SNAPS_PATH" -name '*.snap')
+    fi
+}
+
+download_nested_image(){
+    local IMAGE_URL=$1
+    local IMAGE_NAME=$2
+    local IMAGE_PATH
+    IMAGE_PATH="$(get_image_path)"
+
+    if [[ "$IMAGE_URL" == *.img.xz ]]; then
+        curl -L -o "${IMAGE_PATH}/${IMAGE_NAME}.xz" "$IMAGE_URL"
+        unxz "${IMAGE_PATH}/${IMAGE_NAME}.xz"
+    elif [[ "$IMAGE_URL" == *.img ]]; then
+        curl -L -o "${IMAGE_PATH}/${IMAGE_NAME}" "$IMAGE_URL"
```

Comment:
Could this be simplified to an unconditional `curl -L -o "${IMAGE_PATH}/${IMAGE_NAME}" "$IMAGE_URL"`, and then just unxz ... if url ends with .xz?

Reply:
sure, the only consideration is that unxz just works when the file ends with .xz.
Thanks for the review